test/integration/targets/ec2_instance/roles/ec2_instance/tasks/termination_protection.yml

```diff
@@ -1,101 +1,180 @@
-    ec2_instance:
-      state: absent
-      name: "{{ resource_prefix }}-test-protected-instance-in-vpc"
-      image_id: "{{ ec2_ami_image }}"
-      tags:
-        TestId: "{{ ec2_instance_tag_TestId }}"
-      security_groups: "{{ sg.group_id }}"
-      vpc_subnet_id: "{{ testing_subnet_b.subnet.id }}"
-      termination_protection: true
-      instance_type: "{{ ec2_instance_type }}"
-    register: bad_terminate
-    ignore_errors: yes
-
-  - name: "Cannot terminate protected instance"
-    assert:
-      that:
-        - bad_terminate is failed
-
-  - name: "Alter termination protection setting"
-    ec2_instance:
-      state: present
-      name: "{{ resource_prefix }}-test-protected-instance-in-vpc"
-      image_id: "{{ ec2_ami_image }}"
-      vpc_subnet_id: "{{ testing_subnet_b.subnet.id }}"
-      termination_protection: false
-      instance_type: "{{ ec2_instance_type }}"
-      tags:
-        TestId: "{{ ec2_instance_tag_TestId }}"
-
-  - name: "Try to terminate the instance again (should work)"
-    ec2_instance:
-      state: absent
-      name: "{{ resource_prefix }}-test-protected-instance-in-vpc"
-      image_id: "{{ ec2_ami_image }}"
-      vpc_subnet_id: "{{ testing_subnet_b.subnet.id }}"
-      instance_type: "{{ ec2_instance_type }}"
-      wait: false
-      tags:
-        TestId: "{{ ec2_instance_tag_TestId }}"
-    register: terminate_results
-
-  - assert:
-      that: terminate_results is not failed
+
+    - name: Create instance with termination protection (check mode)
+      ec2_instance:
+        name: "{{ resource_prefix }}-termination-protection"
+        image_id: "{{ ec2_ami_image }}"
+        tags:
+          TestId: "{{ resource_prefix }}"
+        security_groups: "{{ sg.group_id }}"
+        vpc_subnet_id: "{{ testing_subnet_b.subnet.id }}"
+        termination_protection: true
+        instance_type: "{{ ec2_instance_type }}"
+      check_mode: yes
+      register: create_instance_check_mode_results
+
+    - name: Check the returned value for the earlier task
+      assert:
+        that:
+          - "{{ create_instance_check_mode_results.changed }}"
+          - "{{ create_instance_check_mode_results.spec.DisableApiTermination }}"
+
+    - name: Create instance with termination protection
+      ec2_instance:
+        name: "{{ resource_prefix }}-termination-protection"
+        image_id: "{{ ec2_ami_image }}"
+        tags:
+          TestId: "{{ resource_prefix }}"
+        security_groups: "{{ sg.group_id }}"
+        vpc_subnet_id: "{{ testing_subnet_b.subnet.id }}"
+        termination_protection: true
+        instance_type: "{{ ec2_instance_type }}"
+        state: running
+        wait: yes
+      register: create_instance_results
+
+    - name: Check return values of the create instance task
+      assert:
+        that:
+          - "{{ create_instance_results.instances | length }} > 0"
+          - "'{{ create_instance_results.instances.0.state.name }}' in ['running', 'pending']"
```

Comment:
Shouldn't it only be 'running' since wait was True?

Reply:
yes, it should be just running. Also noticed that the check mode tasks don't have `wait: yes`. So I will add that in as well.

Reply:
Hi @s-hertel. I have updated this so that it now checks for the `running` state.